File: api/folder_move_all_versions.go
package vaku

import (
	"context"
	"errors"
)

var (
	// ErrFolderMoveAllVersions when FolderMoveAllVersions fails.
	ErrFolderMoveAllVersions = errors.New("folder move all versions")
)

// FolderMoveAllVersions moves all versions of all secrets from source folder to destination folder.
// Only works on v2 kv engines for both source and destination.
func (c *Client) FolderMoveAllVersions(ctx context.Context, src, dst string) error {
	err := c.FolderCopyAllVersions(ctx, src, dst)
	if err != nil {
		return newWrapErr("", ErrFolderMoveAllVersions, err)
	}

	err = c.FolderDeleteMeta(ctx, src)
	if err != nil {
		return newWrapErr(src, ErrFolderMoveAllVersions, err)
	}

	return nil
}

File: src/api/folder_move_all_versions.rs
//! `FolderMoveAllVersions` — folder_copy_all_versions + folder_delete_meta.

use crate::api::client::Client;
use crate::api::error::{Error, ErrorKind};

impl Client {
    /// Recursive all-versions move (copy then wipe metadata). KV v2 only.
    pub async fn folder_move_all_versions(&self, src: &str, dst: &str) -> Result<(), Error> {
        self.folder_copy_all_versions(src, dst)
            .await
            .map_err(|e| Error::wrap("", ErrorKind::FolderMoveAllVersions, Some(Box::new(e))))?;
        self.folder_delete_meta(src)
            .await
            .map_err(|e| Error::wrap(src, ErrorKind::FolderMoveAllVersions, Some(Box::new(e))))
    }
}
